autoload/build.vim

```diff
@@ -157,25 +236,41 @@ function! s:get_buildsys_item(bs_name, key) " {{{
   endif
 endfunction " }}}
 
-" Return a dictionary containing all fallback targets for the specified language. Will return an
-" empty dictionary if no fallback targets exist. Both s:language_cmds and g:build#languages are
-" considered, while the latter has a higher priority.
+" Return a dictionary containing all available commands for the specified
+" build system. Will return an empty dictionary if nothing is found.
+" If the build_system is not a fallback, both g:build#systems and
+" s:build#systems are considered, the former overwriting the latter.
+" It the build_system is a fallback, both g:build#languages and s:language_cmds
+" are considered.
 "
 " Example:
-"   s:gather_fallback_targets('c')
+"   s:gather_commands({'name': 'c', 'fallback': v:true))
```

Comment:
It is a `build_system` dictionary as returned by `build#get_current_build_system`. I can change that but for the caller it seems easier to use the full dict and it may be better if `gather_commands` needs more info on the build system.